js/summary.js

```diff
@@ -179,9 +182,11 @@ H5P.Summary = (function ($, Question, XApiEventBuilder, StopWatch) {
         that.dataBitMap[panelIndex] = this.dataBitMap[panelIndex] || [];
         that.dataBitMap[panelIndex][id] = summaryIndex;
 
+        emptyEntriesCount = that.dataBitMap[panelIndex].length - (that.dataBitMap[panelIndex].filter(function () { return true; })).length;
```

Comment:
@otacke : Can you please explain this one? I.e: I don't understand what it does, since the filter callback function always returns true? Won't emptyEntriesCount always be zero? Independent of that, I would split this into several lines.

Reply:
@fnoks The `that.dataBitMap[panelIndex]` array can contain `empty` values, which are counted when using `length`, but here we need the number of actual values. `filter` will only work on real values, not on `empty` values, so the `emptyEntriesCount` will not always be zero but the count of the `empty` values.

I have added some verbosity ...